packages/lix-sdk/src/branch/merge-branch.test.ts

```diff
@@ -0,0 +1,255 @@
+				entity_id: "entity1",
+				file_id: "file1",
+				plugin_key: "mock-plugin",
+				snapshot_id: "no-content",
+			},
+			{
+				id: "change2",
+				type: "file",
+				entity_id: "entity2",
+				file_id: "file2",
+				plugin_key: "mock-plugin",
+				snapshot_id: "no-content",
+			},
+			{
+				id: "change3",
+				type: "file",
+				entity_id: "entity3",
+				file_id: "file3",
+				plugin_key: "mock-plugin",
+				snapshot_id: "no-content",
+			},
+		])
+		.returningAll()
+		.execute();
+
+	await lix.db
+		.insertInto("branch_change_pointer")
+		.values([
+			{
+				branch_id: sourceBranch.id,
+				change_id: change1!.id,
+				change_entity_id: change1!.entity_id,
+				change_file_id: change1!.file_id,
+				change_type: change1!.type,
+			},
+			{
+				branch_id: sourceBranch.id,
+				change_id: change2!.id,
+				change_entity_id: change2!.entity_id,
+				change_file_id: change2!.file_id,
+				change_type: change2!.type,
+			},
+			{
+				branch_id: sourceBranch.id,
+				change_id: change3!.id,
+				change_entity_id: change3!.entity_id,
+				change_file_id: change3!.file_id,
+				change_type: change3!.type,
+			},
+		])
+		.execute();
+
+	const mockPlugin: LixPlugin = {
+		key: "mock",
+		detectConflictsV2: async ({ changes }) => {
+			// simulating a conflict with change2
+			return changes
+				.filter((change) => change.id === change2!.id)
+				.map((change) => ({
+					change_id: change.id,
+					// conflicts with itself for simplicity
+					conflicting_change_id: change.id,
+				}));
+		},
+	};
+
+	lix.plugin.getAll = vi.fn().mockResolvedValue([mockPlugin]);
+
+	// Execute the mergeBranch function
+	await mergeBranch({ lix, sourceBranch, targetBranch });
+
+	// Validate results in `branch_change_pointer` and `conflict` tables
+	const targetPointers = await lix.db
+		.selectFrom("branch_change_pointer")
+		.selectAll()
+		.where("branch_id", "=", targetBranch.id)
+		.execute();
+
+	const conflicts = await lix.db.selectFrom("conflict").selectAll().execute();
+
+	// Ensure that non-conflicting pointers (change1 and change3) are in target branch
+	expect(targetPointers.map((pointer) => pointer.change_id)).toContain(
+		change1?.id,
+	);
+	expect(targetPointers.map((pointer) => pointer.change_id)).toContain(
+		change3?.id,
+	);
+
+	// Ensure that conflicting pointer (change2) is not in target branch
+	expect(targetPointers.map((pointer) => pointer.change_id)).not.toContain(
+		change2?.id,
+	);
+
+	// Verify that a conflict for change2 was added to the `conflict` table
+	expect(conflicts.map((conflict) => conflict.change_id)).toContain(
+		change2?.id,
+	);
+});
+
+// it is reasonable to assume that a conflict exists if the same (entity, file, type) change is updated in both branches.
+// in case a plugin does not detect a conflict, the system should automatically detect it.
+test("it should automatically detect a conflict if a change exists that differs updates in both branches despite having a common ancestor", async () => {
```

Comment:
this is a beautiful change. 

plugins do not need to detect diverging entity graphs as conflict anymore. lix handles that as it's universal conflict. an analogy with git: git defines entities in text files as rows. if the same row (entity) diverges in two repos, git raises a conflict. same now applies to lix. if two entities diverge, it's a conflict. but in contrast to git, entities can be defined more fine grained e.g. "character", automatically be resolved, and semantic conflicts between entities can be raised. if row 1 and row 2 are diverging, it's not an 'entity conflict' but maybe a semantic one!